code/game/machinery/firealarm.dm

```diff
@@ -348,6 +348,25 @@
 	else
 		set_light(l_power = 0)
 
+// Allows users to examine the state of the thermal sensor
+/obj/machinery/firealarm/examine(mob/user)
+	. = ..()
+	. += "A light on the side indicates the thermal sensor is [detecting ? "enabled" : "disabled"]."
+
+// Allows Silicons to disable thermal sensor
+/obj/machinery/firealarm/BorgCtrlClick(mob/living/silicon/robot/user)
+	if(get_dist(src,user) <= user.interaction_range)
+		AICtrlClick(user)
+		return
+	return ..()
+
+/obj/machinery/firealarm/AICtrlClick(mob/living/silicon/robot/user)
+	if(obj_flags & EMAGGED)
+		to_chat(user, "<span class='warning'>The control circuitry of [src] appears to be malfunctioning.</span>")
```

Comment:
Use the new procs for the span classes now, please.

Reply:
I'll fix them up when I wake up then. Thanks for pointing them out